sentry_sdk/integrations/aiohttp.py

```diff
@@ -234,27 +248,80 @@ async def on_request_start(session, trace_config_ctx, params):
 
         trace_config_ctx.span = span
 
+        if params.url.path == "/graphql":
+            trace_config_ctx.request_headers = params.headers
+
+    async def on_request_chunk_sent(session, trace_config_ctx, params):
+        # type: (ClientSession, SimpleNamespace, TraceRequestChunkSentParams) -> None
+        if not hasattr(params, "url") or not hasattr(params, "method"):
+            # these are missing from params in earlier aiohttp versions
+            return
+
+        if params.url.path == "/graphql" and params.method == "POST":
+            with capture_internal_exceptions():
+                trace_config_ctx.request_body = json.loads(params.chunk)
+
     async def on_request_end(session, trace_config_ctx, params):
         # type: (ClientSession, SimpleNamespace, TraceRequestEndParams) -> None
-        if trace_config_ctx.span is None:
-            return
+        response = params.response
+
+        if trace_config_ctx.span is not None:
+            span = trace_config_ctx.span
+            span.set_http_status(int(response.status))
+            span.set_data("reason", response.reason)
+
+        hub = Hub.current
+        if (
+            response.url.path == "/graphql"
+            and response.method in ("GET", "POST")
+            and response.status == 200
+        ):
+            with hub.configure_scope() as scope:
+                with capture_internal_exceptions():
+                    response_content = await response.json()
```

Comment:
Because we save the response content, maybe we should add an option to the integration to turn off GraphQL instrumentation? (Or we just yolo it and add the option when people start to complain)

Reply:
An option makes sense, will add.

Reply:
To all three integrations.